File: backend-java/src/main/java/com/nfse/saas/models/app/ServiceOrder.java
package com.nfse.saas.models.app;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "service_orders")
public class ServiceOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "contract_id")
    private Contract contract;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "execution_date")
    private LocalDateTime executionDate;

    private Double value;

    private String status = "PENDING";

    @Column(name = "file_path")
    private String filePath;

    @Column(name = "created_at", insertable = false, updatable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "serviceOrder", fetch = FetchType.LAZY)
    private List<Invoice> invoices;

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getTenantId() { return tenantId; }
    public void setTenantId(Long tenantId) { this.tenantId = tenantId; }
    public Contract getContract() { return contract; }
    public void setContract(Contract contract) { this.contract = contract; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public LocalDateTime getExecutionDate() { return executionDate; }
    public void setExecutionDate(LocalDateTime executionDate) { this.executionDate = executionDate; }
    public Double getValue() { return value; }
    public void setValue(Double value) { this.value = value; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public String getFilePath() { return filePath; }
    public void setFilePath(String filePath) { this.filePath = filePath; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public List<Invoice> getInvoices() { return invoices; }
    public void setInvoices(List<Invoice> invoices) { this.invoices = invoices; }
}

File: backend-python/app/models/service_order.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import BaseApp

class ServiceOrder(BaseApp):
    __tablename__ = "service_orders"

    id = Column(Integer, primary_key=True, index=True)
    tenant_id = Column(Integer, index=True)
    contract_id = Column(Integer, ForeignKey("contracts.id"))
    
    description = Column(Text)
    execution_date = Column(DateTime)
    value = Column(Float)
    
    status = Column(String, default="PENDING") # PENDING, EXECUTED, BILLED
    file_path = Column(String, nullable=True) # Path to the uploaded OS file/image
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    # Relationships
    contract = relationship("Contract", back_populates="service_orders")
    invoices = relationship("Invoice", back_populates="service_order")
